app/core/edge_inference.py

```diff
@@ -62,14 +62,48 @@ def parse_inference_response(response: dict) -> dict:
     #     "predictions": {"confidences": [0.54], "labels": [0], "probabilities": [0.45], "scores": [-2.94]},  # Binary results
     #     "secondary_predictions": None,  # Text recognition and Obj detection results
     # }
-    output_dict = response["predictions"]
-    output_dict = {
-        "score": output_dict["scores"][0],
-        "confidence": output_dict["confidences"][0],
-        "probability": output_dict["probabilities"][0],
-        "label": output_dict["labels"][0],
-    }
-    output_dict["label"] = "NO" if output_dict["label"] else "YES"  # map false / 0 to "YES" and true / 1 to "NO"
+
+    multi_predictions: dict = response.get("multi_predictions", None)
+    predictions: dict = response.get("predictions", None)
+    secondary_predictions: dict = response.get("secondary_predictions", None)
+
+    if multi_predictions is not None and predictions is not None:
+        raise ValueError("Got result with both multi_predictions and predictions.")
+    if multi_predictions is not None:
```

Comment:
What's a "multi_prediction"?

Reply:
That's the field where multiclass/counting results are returned through.